From b861c9edbec7478c5dbaafeeb4da0a8fe1c3efa1 Mon Sep 17 00:00:00 2001
From: Kemeng Shi <shikemeng@huawei.com>
Date: Sat, 8 Jan 2022 18:52:25 +0800
Subject: [PATCH] etmem: add ioctl for mm idle scan
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: b861c9edbec7478c5dbaafeeb4da0a8fe1c3efa1


euleros inclusion
category: feature
feature: etmem
bugzilla: https://gitee.com/openeuler/kernel/issues/I4OODH?from=project-issue
CVE: NA

-------------------------------------------------

support ioctl for etmem scan to set scan flag

Signed-off-by: Kemeng Shi <shikemeng@huawei.com>
Reviewed-by: louhongxiang <louhongxiang@huawei.com>
Reviewed-by: Chen Wandun <chenwandun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/proc/task_mmu.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/fs/proc/task_mmu.c b/fs/proc/task_mmu.c
index 4da09b397370..2d9510cf30c3 100644
--- a/fs/proc/task_mmu.c
+++ b/fs/proc/task_mmu.c
@@ -1890,11 +1890,20 @@ static int mm_idle_release(struct inode *inode, struct file *file)
 	return 0;
 }
 
+static long mm_idle_ioctl(struct file *filp, unsigned int cmd, unsigned long arg)
+{
+	if (proc_page_scan_operations.unlocked_ioctl)
+		return proc_page_scan_operations.unlocked_ioctl(filp, cmd, arg);
+
+	return 0;
+}
+
 const struct file_operations proc_mm_idle_operations = {
 	.llseek		= mem_lseek, /* borrow this */
 	.read		= mm_idle_read,
 	.open		= mm_idle_open,
 	.release	= mm_idle_release,
+	.unlocked_ioctl = mm_idle_ioctl,
 };
 
 /*swap pages*/
-- 
2.26.2

